packages/console/src/pages/TenantSettings/Subscription/ConsoleEmbedPricing/index.tsx

```diff
@@ -0,0 +1,157 @@
+import { condString, joinPath } from '@silverhand/essentials';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import { consoleEmbedPricingLink } from '@/consts/env';
+import Card from '@/ds-components/Card';
+import useTheme from '@/hooks/use-theme';
+import useUserPreferences from '@/hooks/use-user-preferences';
+
+import Skeleton from './Skeleton';
+import styles from './index.module.scss';
+
+const websiteSender = 'website_console_embed_pricing';
+const sender = 'console_pricing';
+
+enum MessageType {
+  RequestContentHeight = 'requestContentHeight',
+  ReportContentHeight = 'reportContentHeight',
+  ChangeTheme = 'changeTheme',
+}
+
+type MessageData = {
+  sender: string;
+  type: MessageType;
+  payload?: {
+    theme?: 'light' | 'dark';
+    height?: number;
+  };
+};
+
+function ConsoleEmbedPricing() {
+  const [height, setHeight] = useState<number | undefined>(window.innerHeight);
+  const pricingContentRef = useRef<HTMLIFrameElement>(null);
+  const [iframeLoaded, setIframeLoaded] = useState(false);
+
+  const theme = useTheme();
+  const {
+    data: { language },
+  } = useUserPreferences();
+
+  const pricingContentUrl = useMemo(() => {
+    const url = new URL(
+      joinPath(
+        // Ignore language tag if it's English
+        condString(language !== 'en' && language),
+        condString(consoleEmbedPricingLink.path)
+      ),
+      consoleEmbedPricingLink.domain
+    );
+    url.searchParams.set('theme', theme);
+
+    return url;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    /**
+     * Intentionally not including `theme` in the dependency array, the `theme` here is only used for the initial rendering
+     * and will update using the postMessage to avoid page reload when changing the theme.
+     */
+    language,
+  ]);
```

Comment:
nit: Should we consider a fallback to English when the console language is not supported by the official website? I know this is an extreme case and seems unlikely to happen, but just in case?

Reply:
The website has already handled this issue.